CMakeLists.txt

```diff
@@ -49,8 +49,20 @@ endif()
 if(NOT CMAKE_VERSION VERSION_LESS 3.4.0 AND NOT CMAKE_SYSTEM_NAME STREQUAL "Android")
     set(LANG CXX)
 endif()
+
+
+option(HUNTER_ENABLED "Get third parties (Corrade, glfw, SDL2) via hunter " OFF)
+if (HUNTER_ENABLED)
+    set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_LIST_DIR}/package/hunter")
+    include(HunterInit)
+endif()
```

Comment:
As far as I can see, Corrade did not need any patching for Hunter support because 

- it doesn't have any dependencies
- thus no `hunter_add_package()` needs to be called
- thus there's no need to include HunterGate

However, here because we have dependencies and we need a bunch of `hunter_add_package()` for them, we need to include HunterGate. Is that correct?

I was thinking that, since Hunter will be *very seldom* used to develop Magnum itself and it will be mostly only used to develop projects *depending on* Magnum, we could skip including HunterGate and have just the `hunter_add_package()` calls wrapped in `if(HUNTER_ENABLED)`, assuming HunterGate would be already included from some super-project. But I could not find any concrete specification of this in the docs so I'm not sure if this is at all possible. Also, not sure about the behavior when Hunter CIs are producing binary versions of Magnum. So we *have to* include HunterGate in any case. Right?

Reply:
> As far as I can see, Corrade did not need any patching for Hunter support because
> it doesn't have any dependencies
> thus no hunter_add_package() needs to be called
> thus there's no need to include HunterGate

That is right

> However, here because we have dependencies and we need a bunch of hunter_add_package() for them, we need to include HunterGate. Is that correct?
I do not think we need to include HunterGate into corrade : it does not need to do any call to `hunter_add_package`.

Concerning magnum, yes we do need it. 


> I was thinking that, since Hunter will be very seldom used to develop Magnum itself and it will be mostly only used to develop projects depending on Magnum, we could skip including HunterGate and have just the hunter_add_package() calls wrapped in if(HUNTER_ENABLED), assuming HunterGate would be already included from some super-project. But I could not find any concrete specification of this in the docs so I'm not sure if this is at all possible

I do not think that it is possible and I would not recommend it.

Let me explain: this inclusion is critical in order to be able to test the hunter build locally (and not wait for travis to fail).

Let me show you how the magnum_hunter repo can be used to test the magnum build in an environment that reproduces travis configuration. 

Step 1 : launch a docker image that reproduces travis build config:
````bash
> cd /path/to/magnum_hunter/Docker/ubuntu
➜  ubuntu git:(master) ✗> ./docker_create_image.sh  # Create a docker image close to travis CI config
>  ubuntu git:(master) ✗ ./docker_login.sh # log inside the container (beware, all modifs are removed after logout)
`````

Step 2 : you are now inside a docker container, where /sources_docker <=> /path/to/magnum_hunter

Example 1 : build magnum : this will build corrade via hunter, and magnum the standard way, using the toolchain gcc-7-cxx14

`````bash
➜  /sources_docker git:(master) ✗ ./TLDR_hunter.py test-build magnum gcc-7-cxx14
`````

Example 2 : build magnum_example_app_with_submodules. Since this project uses hunter from submodules , it will test the packaging of the current versions of magnum and corrade.
````bash
corrade and magnum versions (+ the packaging of sdl, glfw, etc)
➜  /sources_docker git:(master) ✗ ./TLDR_hunter.py test-build magnum_example_app_with_submodules gcc-7-cxx14
````

This could be extended to magnum plugins later.

Reply:
I see, thanks for the extensive explanation.

Reply:
@ruslo Yes, I'm aware of that ;) This `if()` was added in order to not require `HunterGate.cmake` to be present in the source directory at all, which I requested.